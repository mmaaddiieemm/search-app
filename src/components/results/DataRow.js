import React from "react"; 
import VerifyData from "../../VerifyData.js"; 

// return a single formatted row of data given the expected keys
// for that row.  
function DataRow( {data, keys}) {
    return (
      <>
            {keys.map((item) => {
                
                return(< label key={item}><b> {item}: </b> {VerifyData(data, item)} </label>)
            })}
      </>
    ); 
  }
  
  export default DataRow; 