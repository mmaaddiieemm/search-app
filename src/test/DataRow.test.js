import {DataRow, verifyData} from "../DataRow"; 

test('When undefined or null data is passed in to verifyData, return undefined', () => {
    const result = verifyData(null, 'toots'); 
    
}); 

test('when the parameter passed in is unefined, null or not in the data, render undefined', () => {

}); 

test('DataRow renders the number of items specified in the keys parameter', () => {

}); 

test('When data at param is valid, return the data in a string form', () => {
    const testData = {
        nameThing: 'a_unique_name'
    }; 
    const result = verifyData(testData, 'nameThing'); 
    expect(result).toBeStrictlyEqual('a_unique_name'); 
})