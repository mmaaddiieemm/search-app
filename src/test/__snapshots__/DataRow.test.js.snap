// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When data at param is valid, return the data in a string form 1`] = `
Array [
  <label>
    <b>
       
      nameThing
      :
       
    </b>
     
    a_unique_name
     
  </label>,
  <label>
    <b>
       
      arrayThing
      :
       
    </b>
     
    [name1,another thing]
     
  </label>,
]
`;

exports[`When undefined or null data is passed in to verifyData 1`] = `
<label>
  <b>
     
    nameThing
    :
     
  </b>
   
  a_unique_name
   
</label>
`;
